sql/dd/upgrade_57/table.cc

```diff
@@ -1598,6 +1628,15 @@ static bool migrate_table_to_dd(THD *thd, const String_type &schema_name,
       key_info->is_algorithm_explicit = true;
   }
 
+  // if table is part of tablespace - check if tablespace is encrypted
+  // if it is - assing encryption clause to the table
+  if (((share.tablespace && is_tablespace_encrypted(share.tablespace)) ||
+       (schema_name == "mysql" && bootstrap::DD_bootstrap_ctx::instance()
+                                      .is_system_tablespace_encrypted())) &&
+      !is_encrypted(share.encrypt_type)) {
+    share.encrypt_type.str = C_STRING_WITH_LEN("Y");
```

Comment:
Shouldn't this be done at InnoDB layer? the table attributes can be fixed/updated in one of DD upgrade APIs?

Reply:
This is for upgrade from 5.7. In this function .frm is read and based on the information from .frm CREATE_INFO is created - that should contain encrypt_type information. Function name:
/**
  Read .frm files and enter metadata for tables/views.
*/

static bool migrate_table_to_dd(THD *thd, const String_type &schema_name,

Reply:
reading .frms doesn't mean it is interacting with InnoDB. Please consider doing this in InnoDB. Specially the dd_upgrade_table() API.

It should be simpler too
something like 
```
if (table->space == 0 && srv_sys_space.is_encrypted() (or some flags check))
 {
       add encryption attribute
}
```

Reply:
slight change of mind:

during upgrade, server has to add "encrypt_type=n" right for all the tables upgraded from 5.7. Where does this happen? if it happens in server, am OK if you can do at same place.

Reply:
The code there is not needed. This is because sql/dd/impl/upgrade/dd.cc:update_meta_data is already updating table's encrypt_type based on general tablespace encrypt_type. Removing.